plugins/main/macvlan/macvlan.go

```diff
@@ -107,12 +107,35 @@ func loadConf(args *skel.CmdArgs, envArgs string) (*NetConf, string, error) {
 	if err := json.Unmarshal(args.StdinData, n); err != nil {
 		return nil, "", fmt.Errorf("failed to load netconf: %v", err)
 	}
-	if n.Master == "" {
-		defaultRouteInterface, err := getNamespacedDefaultRouteInterfaceName(args.Netns, n.LinkContNs)
+
+	var result *current.Result
+	var err error
+	// Parse previous result
+	if n.NetConf.RawPrevResult != nil {
+		if err = version.ParsePrevResult(&n.NetConf); err != nil {
+			return nil, "", fmt.Errorf("could not parse prevResult: %v", err)
+		}
+
+		result, err = current.NewResultFromResult(n.PrevResult)
 		if err != nil {
-			return nil, "", err
+			return nil, "", fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	if n.Master == "" {
+		if result == nil {
+			var defaultRouteInterface string
+			defaultRouteInterface, err = getNamespacedDefaultRouteInterfaceName(args.Netns, n.LinkContNs)
+			if err != nil {
+				return nil, "", err
+			}
+			n.Master = defaultRouteInterface
+		} else {
+			if len(result.Interfaces) == 1 && result.Interfaces[0].Name != "" {
```

Comment:
Rather than just picking the first interface, what if we were a bit cleverer?

Right now, the macvlan plugin has the option to pick from the host namespace or container namespace. Let's preserve that when it is chained. Specifically, if `n.LinkContNS` is true, then pick the first value in `prevResult` where `sandbox == true`. Even better would be to pick the first result with routes or an IP address. Likewise, if `n.LinkContNS` is false, then do the same but for host-side interfaces.

Makes sense?

Reply:
@mmirecki any thoughts here?

Reply:
> Rather than just picking the first interface, what if we were a bit cleverer?
> 
> Right now, the macvlan plugin has the option to pick from the host namespace or container namespace. Let's preserve that when it is chained. Specifically, if `n.LinkContNS` is true, then pick the first value in `prevResult` where `sandbox == true`. Even better would be to pick the first result with routes or an IP address. Likewise, if `n.LinkContNS` is false, then do the same but for host-side interfaces.
> 
> Makes sense?

We have a scenario where we need to dynamically select the host master to create macvlan, and I don't know much about the scenario where we need to configure macvlan in the container.